sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ConditionalPreconditionsTest.java

```diff
@@ -651,6 +657,852 @@ void testDescriptionModificationModificationNoDescriptors() {
+
+        // given
+        final var descriptor1Handle = "superHandle1";
+        final var descriptor2Handle = "handle;Süper;2";
+
+        final var presenceMap = new HashMap<>(Map.of(
+                descriptor1Handle, false,
+                descriptor2Handle, true));
+
+        when(mockManipulations.getRemovableDescriptorsOfClass(MdsDescriptor.class))
+                .thenReturn(List.of(descriptor1Handle, descriptor2Handle));
+
+        when(mockManipulations.insertDescriptor(anyString())).thenAnswer((Answer<ResponseTypes.Result>) invocation -> {
+            presenceMap.put(invocation.getArgument(0), true);
+            return ResponseTypes.Result.RESULT_SUCCESS;
+        });
+        when(mockManipulations.removeDescriptor(anyString()))
+                .thenAnswer((Answer<ResponseTypes.Result>) invocation -> ResponseTypes.Result.RESULT_NOT_SUPPORTED);
+        when(mockManipulations.triggerDescriptorUpdate(anyString()))
+                .thenAnswer((Answer<ResponseTypes.Result>) invocation -> ResponseTypes.Result.RESULT_SUCCESS);
+        when(testClient.getSdcRemoteDevice().getMdibAccess().getDescriptor(anyString()))
+                .thenAnswer((Answer<Optional<AbstractDescriptor>>) invocation -> {
+                    final String handle = invocation.getArgument(0);
+                    if (presenceMap.get(handle)) {
+                        return Optional.of(mock(AbstractDescriptor.class));
+                    } else {
+                        return Optional.empty();
+                    }
+                });
+
+        // when
+        assertFalse(
+                ConditionalPreconditions.DescriptionModificationMdsDescriptorPrecondition.manipulation(testInjector));
+
+        // then
+        final var insertCaptor = ArgumentCaptor.forClass(String.class);
+        final var removeCaptor = ArgumentCaptor.forClass(String.class);
+        final var updateCaptor = ArgumentCaptor.forClass(String.class);
+        verify(mockManipulations, times(1)).getRemovableDescriptorsOfClass(MdsDescriptor.class);
+        verify(mockManipulations, times(1)).insertDescriptor(insertCaptor.capture());
+        verify(mockManipulations, times(2)).removeDescriptor(removeCaptor.capture());
+        verify(mockManipulations, times(2)).triggerDescriptorUpdate(updateCaptor.capture());
+
+        assertEquals(descriptor1Handle, updateCaptor.getAllValues().get(0));
+        assertEquals(descriptor2Handle, updateCaptor.getAllValues().get(1));
+        assertEquals(descriptor1Handle, insertCaptor.getValue());
+        assertEquals(descriptor1Handle, removeCaptor.getAllValues().get(0));
+        assertEquals(descriptor2Handle, removeCaptor.getAllValues().get(1));
+    }
+
+    /**
+     * Tests whether DescriptionModificationMdsDescriptorPrecondition fails when
+     * all Updates fail.
+     */
+    @Test
+    @DisplayName("DescriptionModificationMdsDescriptorPrecondition fails when all Updates fail")
```

Comment:
As above, the update is not successful, but also not failed. This wording could be misleading, since there is also an explicit fail response